subsys/net/ip/tcp.c

```diff
@@ -1226,6 +1259,15 @@ static void tcp_resend_data(struct k_work *work)
 	conn->data_mode = TCP_DATA_MODE_RESEND;
 	conn->unacked_len = 0;
 
+#ifdef CONFIG_NET_TCP_RANDOMIZED_RTO
+	/* If the retransmit was hit for the first time,
+	 * compute a new retransmit timeout, for the next retransmit
+	 */
+	if (conn->send_data_retries == 0) {
```

Comment:
Please change this to
`if (IS_ENABLED(CONFIG_NET_TCP_RANDOMIZED_RTO) && conn->send_data_retries == 0) {`
so we can remove the above ifdef check.

Reply:
Good suggestion, was not aware of this construction